### PR TITLE
Indentifier: class to get identifier hash bucket

### PR DESCRIPTION
**Pull request by DFoxinator**
@be-dmitry 
@ingluisjimenez 
@amy 

Proof using this script:
```
$buckets = [];

// 1,000,000 identifiers
for ( $i = 0; $i < 1000000; ++$i ) {
   $bucket = crc32('feature_name_' . mt_rand(1, 999999999999999));
   $bucket = ($bucket % 100) + 1;

   if (!isset($buckets[$bucket])) {
      $buckets[$bucket] = 0;
   }

    ++$buckets[$bucket];
}
```

Results from run on 1,000,000,000 identifiers (bucket #, # of identifiers in bucket):
```
[1] => 10003505
[2] => 10001727
[3] => 9999428
[4] => 9994767
[5] => 9998631
[6] => 10002999
[7] => 9994422
[8] => 9996041
[9] => 10001584
[10] => 10007482
[11] => 10000600
[12] => 10000428
[13] => 9998448
[14] => 9998182
[15] => 9999516
[16] => 10002951
[17] => 9998210
[18] => 9993653
[19] => 9997865
[20] => 9999021
[21] => 10001587
[22] => 10001981
[23] => 9998335
[24] => 10003767
[25] => 9995172
[26] => 9989743
[27] => 10008791
[28] => 10003549
[29] => 9999336
[30] => 10000029
[31] => 9991081
[32] => 10003796
[33] => 9993984
[34] => 9998345
[35] => 9994354
[36] => 9999280
[37] => 10004024
[38] => 10010801
[39] => 10001638
[40] => 9998344
[41] => 9997133
[42] => 10002074
[43] => 10003359
[44] => 10001069
[45] => 10002989
[46] => 10000607
[47] => 10004450
[48] => 10004440
[49] => 9999632
[50] => 10000390
[51] => 10002930
[52] => 10000480
[53] => 9996053
[54] => 9998635
[55] => 9995792
[56] => 9998402
[57] => 9999403
[58] => 10002926
[59] => 10003008
[60] => 9998575
[61] => 10002408
[62] => 10005427
[63] => 9996286
[64] => 9995589
[65] => 10000834
[66] => 10000396
[67] => 10003539
[68] => 10001571
[69] => 10002071
[70] => 10001913
[71] => 10003955
[72] => 10002976
[73] => 10002813
[74] => 10002665
[75] => 9998182
[76] => 10003885
[77] => 9993679
[78] => 9997372
[79] => 9991577
[80] => 10001174
[81] => 9998502
[82] => 9999236
[83] => 9999025
[84] => 9999211
[85] => 10004182
[86] => 9994243
[87] => 10000994
[88] => 10003862
[89] => 10005745
[90] => 9993825
[91] => 9995505
[92] => 9995910
[93] => 10004004
[94] => 9998491
[95] => 10000452
[96] => 9989992
[97] => 10000457
[98] => 10001010
[99] => 9998823
[100] => 10008475
```

### CODE REVIEW

File: src/IdentifierHashBucket.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    return $this->_bucket;
+
+  } // getBucket
+
+  protected function _calculateBucket() {
+
+    $hash          = crc32( $this->_feature . $this->_identifier );
+    $this->_bucket = ($hash % $this->_num_buckets) + 1;
```
Comment (be-dmitry) [inline src/IdentifierHashBucket.php]: php sniffs here

File: tests/unit/IdentifierHashBucketTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  } // getBucketDefault
+
+  /**
+   * @return array
+   */
+  public function getBucketDefaultProvider() {
+
+    return [
+        [ '234234788485', 43, 36 ],
```
Comment (be-dmitry) [inline tests/unit/IdentifierHashBucketTest.php]: Try adding some alphanumeric identifiers, like an adobe id

Comment (DFoxinator) [inline tests/unit/IdentifierHashBucketTest.php]: Good idea

File: src/IdentifierHashBucket.php (hunk 1 above)
Comment (DFoxinator) [inline src/IdentifierHashBucket.php]: The sniffs are turned on, is this definitely a rule?

Comment (be-dmitry) [inline src/IdentifierHashBucket.php]: I thought it was. Maybe not. Anyhow, carry on

Comment (DFoxinator) [inline src/IdentifierHashBucket.php]: I changed it anyway, looks a bit cleaner.

**Comment by amy** [comment]
👍

**Comment by be-dmitry** [comment]
👍

**Pull request merged**